src/DIRAC/Core/Tornado/Server/BaseRequestHandler.py

```diff
@@ -0,0 +1,739 @@
+      self.write(encode(self.result))
+
+    self.finish()
+
+  def _parseDIRACResult(self, result):
+    """ Processing of a standard DIRAC result,
+        but in a separate method so that it can be modified for another class if necessary
+    """
+    self.set_header("Content-Type", "application/json")
+    self.write(encode(result))
+
+  def on_finish(self):
+    """
+      Called after the end of HTTP request.
+      Log the request duration
+    """
+    elapsedTime = 1000.0 * self.request.request_time()
+
+    argsString = "OK"
+    try:
+      if not self.result['OK']:
+        argsString = "ERROR: %s" % self.result['Message']
+    except (AttributeError, KeyError, TypeError):  # In case it is not a DIRAC structure
+      if self._reason != 'OK':
+        argsString = 'ERROR %s' % self._reason
+
+    sLog.notice("Returning response", "%s %s (%.2f ms) %s" % (self.srv_getFormattedRemoteCredentials(),
+                                                              self._serviceName,
+                                                              elapsedTime, argsString))
+
+  def _gatherPeerCredentials(self, grants=None):
+    """ Returne a dictionary designed to work with the AuthManager,
+        already written for DISET and re-used for HTTPS.
+
+        :param list grants: grants to use
+
+        :returns: a dict containing the return of :py:meth:`DIRAC.Core.Security.X509Chain.X509Chain.getCredentials`
+                  (not a DIRAC structure !)
+    """
+    err = []
+    result = None
+
+    grants = grants or self.USE_AUTHZ_GRANTS
+
+    if not grants:
+      raise Exception('USE_AUTHZ_GRANTS is not defined.')
+
+    for a in grants:
+      grant = a.upper()
+      grantFunc = eval('self._authz%s' % grant)
+      if not callable(grantFunc):
+        raise Exception('%s authentication type is not supported.' % grant)
+      result = grantFunc()
+      if result['OK']:
+        for e in err:
+          sLog.debug(e)
+        sLog.debug('%s authentication success.' % grant)
+        return result['Value']
+      err.append('%s authentication: %s' % (grant, result['Message']))
+
+    # Report on failed authentication attempts
+    raise Exception('; '.join(err))
+
+  def _authzSSL(self):
+    """ Load client certchain in DIRAC and extract informations.
+
+        :return: S_OK(dict)/S_ERROR()
+    """
+    peerChain = X509Chain()
+    derCert = self.request.get_ssl_certificate()
+
+    # Get client certificate pem
+    if derCert:
+      chainAsText = derCert.as_pem()
+      # Here we read all certificate chain
+      cert_chain = self.request.get_ssl_certificate_chain()
+      for cert in cert_chain:
+        chainAsText = cert.as_pem()
+    elif self.request.headers.get('X-Ssl_client_verify') == 'SUCCESS':
+      chainAsTextEncoded = self.request.headers.get('X-SSL-CERT')
+      chainAsText = unquote(chainAsTextEncoded)
+    else:
+      return S_ERROR('Not found a valide client certificate.')
```

Comment:
```suggestion
      return S_ERROR('Valid certificate not found')
```
but please use `DErrno`.

Reply:
added